packages/govuk-frontend/src/govuk/components/accordion/accordion.test.js

```diff
@@ -733,6 +733,45 @@ describe('/components/accordion', () => {
               message: 'GOV.UK Frontend is not supported in this browser'
             })
           })
+
+          it('throws when $module is not set', async () => {
+            await expect(
+              renderAndInitialise(page, 'accordion', {
+                params: examples.default,
+                beforeInitialisation() {
+                  // Remove the root of the components as a way
+                  // for the constructor to receive the wrong type for `$module`
+                  document.querySelector('[data-module]').remove()
+                }
+              })
+            ).rejects.toEqual({
+              name: 'MissingElementError',
+              message: 'Accordion: $module not found'
+            })
+          })
+
+          it('throws when receiving the wrong type for $module', async () => {
+            await expect(
+              renderAndInitialise(page, 'accordion', {
+                params: examples.default,
+                beforeInitialisation() {
+                  // Remove the root of the components for replacement
+                  document.querySelector('[data-module]').remove()
+
+                  // And replace with an `<svg>` element which doesn't have the right type
+                  const $svg = document.createElementNS(
+                    'http://www.w3.org/2000/svg',
+                    'svg'
+                  )
+                  $svg.setAttribute('data-module', 'accordion')
+                  document.body.appendChild($svg)
+                }
```

Comment:
Perhaps the same here?

Just a quick `outerHTML` over the top of `$module` would be great for tests

```suggestion
                beforeInitialisation: ($module) {
                  $module.outerHTML = '<svg data-module="accordion"></svg>'
                }
```

Reply:
If it's helpful, can always add another `@param` like `moduleName`?

```suggestion
                beforeInitialisation($module, moduleName) {
                  $module.outerHTML = `<svg data-module="${moduleName}"></svg>`
                }
```

Reply:
Good shout, didn't think of `outerHTML` for a less tedious replacement, I'll update 😊